src/test/HostActivationTests/NativeHosting/Nethost.cs

```diff
@@ -0,0 +1,167 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.DotNet.Cli.Build.Framework;
+using System;
+using System.IO;
+using System.Runtime.InteropServices;
+using Xunit;
+
+namespace Microsoft.DotNet.CoreSetup.Test.HostActivation.NativeHosting
+{
+    public class Nethost : IClassFixture<Nethost.SharedTestState>
+    {
+        private const string GetHostFxrPath = "nethost_get_hostfxr_path";
+        private const int CoreHostLibMissingFailure = unchecked((int)0x80008083);
+
+        private readonly SharedTestState sharedState;
+
+        public Nethost(SharedTestState sharedTestState)
+        {
+            sharedState = sharedTestState;
+        }
+
+        [Theory]
+        [InlineData(false, true)]
+        [InlineData(false, false)]
+        [InlineData(true, true)]
+        [InlineData(true, false)]
+        public void GetHostFxrPath_DotNetRootEnvironment(bool useAssemblyPath, bool isValid)
+        {
+            string dotNetRoot = isValid ? Path.Combine(sharedState.ValidInstallRoot, "dotnet") : sharedState.InvalidInstallRoot;
+            CommandResult result = Command.Create(sharedState.NativeHostPath, $"{GetHostFxrPath} {(useAssemblyPath ? sharedState.TestAssemblyPath : string.Empty)}")
+                .CaptureStdErr()
+                .CaptureStdOut()
+                .EnvironmentVariable("DOTNET_ROOT", dotNetRoot)
+                .Execute();
+
+            if (isValid)
+            {
+                result.Should().Pass()
+                    .And.HaveStdOutContaining($"hostfxr_path: {sharedState.HostFxrPath}".ToLower());
+            }
+            else
+            {
+                result.Should().Fail()
+                    .And.ExitWith(1)
+                    .And.HaveStdOutContaining($"{GetHostFxrPath} failed: 0x{CoreHostLibMissingFailure.ToString("x")}");
```

Comment:
You could additionally validate that the error written out is expected  - through enabling tracing.